src/Symfony/Component/Routing/Attribute/AsRoutingService.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Attribute;
+
+#[\Attribute(\Attribute::TARGET_CLASS)]
+class AsRoutingService
+{
+    public function __construct(
+        public string $name,
```

Comment:
The name can be optional too as Symfony should fallback to the service-id (to be confirmed ;) )